services/tracing.py

```diff
@@ -93,24 +101,19 @@ def _set_attributes(span, attributes_dict):
         @wraps(func)
         def wrap_with_span(*args, **kwargs):
             name = span_name or func.__qualname__
+            attributes_dict = attributes or {}
+            if func_attributes is not None:
+                func_signature = signature(func)
```

Comment:
Can we move this line up into the decorator body? Then its called once at import time, rather than on every func invocation?